pkg/experimentcalculator/experimentcalc/experiment_calculator.go

```diff
@@ -347,26 +346,29 @@ func (e ExperimentCalculator) appendVariationResult(
 	goalResult *eventcounter.GoalResult,
 	srcVrs []*eventcounter.VariationResult,
 ) {
-	sort.SliceStable(goalResult.VariationResults, func(i, j int) bool {
-		return goalResult.VariationResults[i].VariationId < goalResult.VariationResults[j].VariationId
-	})
-	sort.SliceStable(srcVrs, func(i, j int) bool {
-		return srcVrs[i].VariationId < srcVrs[j].VariationId
-	})
 	for i := 0; i < len(goalResult.VariationResults); i++ {
-		goalResult.VariationResults[i].ExperimentCount = copyVariationCount(srcVrs[i].ExperimentCount)
-		goalResult.VariationResults[i].EvaluationCount = copyVariationCount(srcVrs[i].EvaluationCount)
+		variationResult := getVariationResult(srcVrs, goalResult.VariationResults[i].VariationId)
+		if variationResult == nil {
```

Comment:
Why do you think this could be "true"?
I just want to know the case that `goalResult` and `srcVrs` have different(or more/less) variation ids 🤔

Reply:
Good question!

You can check these line of code that calculating experiment result:

1. get the variation ids that need to be calculated [[1]](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experimentcalculator/experimentcalc/experiment_calculator.go#L155) [[2]](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experimentcalculator/experimentcalc/experiment_calculator.go#L163-L165)
2. Then we will get goalVariationCounts and evaluationVariationCounts based on variation ids [[1]](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experimentcalculator/experimentcalc/experiment_calculator.go#L174) [[2]](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experimentcalculator/experimentcalc/experiment_calculator.go#L193)
3. Calculate variation results using goalVariationCounts and evaluationVariationCounts [[1]](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experimentcalculator/experimentcalc/experiment_calculator.go#L314-L317)

The whole process will keep `goalResult` and `srcVrs` (Goalresult from `3` step) the same variation ids.